libgpos/src/_api.cpp

```diff
@@ -44,9 +44,17 @@ bool fEnableExtendedAsserts = false;  // by default, extended asserts are disabl
 //		Initialize GPOS memory pool, worker pool and message repository
 //
 //---------------------------------------------------------------------------
-void __attribute__((constructor)) gpos_init()
-{
-	if (GPOS_OK != gpos::CMemoryPoolManager::EresInit())
+void gpos_init(struct gpos_init_params* params) {
```

Comment:
have you considered passing `params` directly by value? on most systems that we care about it's no slower (or indeed it's faster) than pointer passing